pombola/south_africa/static/js/track-profile-analytics.js

```diff
@@ -0,0 +1,92 @@
+$(function () {
+  // If Google Analytics is loaded, try send events when elements are visible in viewport
+  if (typeof ga === 'function') {
+    // The timeout is set to 5 seconds to allow us to track actual engagement
+    const TIMEOUT_BEFORE_FIRING_ANALYTICS = 5000;
+    var elementsToTrackScrollEvents = {
+      "#membersinterests": {
+        "category": "person-page-tab-membersinterests",
+        "action": "activate",
+        "label": "members-interests"
+      },
+      "#member-experience": {
+        "category": "person-page-tab-experience",
+        "action": "activate",
+        "label": "member-experience"
+      },
+      "#apperances": {
+        "category": "person-page-tab-appearances",
+        "action": "activate",
+        "label": "member-apperances"
+      },
+      "#messages-section": {
+        "category": "person-page-tab-messages",
+        "action": "activate",
+        "label": "messages"
+      },
+      "#attendance": {
+        "category": "person-page-tab-attendance",
+        "action": "activate",
+        "label": "attendance"
+      },
+    };
+
+    var elementsToTrackExpandDefinitionEvents = {
+      "#members-interests-section": {
+        "category": "collapsed-definition",
+        "action": "expand",
+        "label": "members-interests"
+      },
+      "#plenary-apperances-section": {
+        "category": "collapsed-definition",
+        "action": "expand",
+        "label": "plenary-apperances"
+      },
+      "#committee-meetings-title": {
+        "category": "collapsed-definition",
+        "action": "expand",
+        "label": "committee-meetings"
+      },
+      "#minister-questions-title": {
+        "category": "collapsed-definition",
+        "action": "expand",
+        "label": "questions-to-ministers"
+      },
+    };
+
+
+    function sendAnalyticsEvent(category, action, label) {
+      // we use a timeout to ensure that the event is sent after user has
+      // scrolled over the element considerably
+      setTimeout(function () {
+        ga('send', {
+          hitType: 'event',
+          eventCategory: category,
+          eventAction: action,
+          eventLabel: label
+        });
+      }, TIMEOUT_BEFORE_FIRING_ANALYTICS);
+    }
+
```

Comment:
looks like this will send it 5s after it scroleld into view. I think what we wanted was to send if it was in view _for 5s_.

Reply:
this is the idea - cumulative is fine https://support.google.com/tagmanager/answer/7679410?hl=en#advanced

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API/Timing_element_visibility